examples/src/main/java/com/cloudera/nav/sdk/examples/lineage3/LineageExport.java

```diff
@@ -0,0 +1,219 @@
+    this.lineageOptions = lineageOptions;
+  }
+
+  public void run() throws IOException {
+    NavApiCient client = NavigatorPlugin.fromConfigFile(configPath)
+        .getClient();
+
+    String cursorMark = null;
+    List<Map<String, Object>> results = null;
+    ResultsBatch<Map<String, Object>> rb;
+    FileWriter writer = new FileWriter(DEFAULT_FILENAME);
+
+    do {
+      MetadataQuery mq = new MetadataQuery(query, 100, cursorMark);
+      rb = client.getEntityBatch(mq);
+
+      results = rb.getResults();
+      cursorMark = rb.getCursorMark();
+
+      // extract out all the entityIds for which to download
+      // lineage for
+      Set<String> entityIds = Sets.newHashSet();
+      for (Map<String, Object> entities : results) {
+        entityIds.add((String) entities.get(IDENTITY));
+      }
+
+      if (entityIds.size() > 0) {
+        fetchAndWriteLineage(entityIds, writer, client.getConfig());
+      }
+    } while (results.size() != 0);
+
+    writer.flush();
+    writer.close();
+  }
+
+  @SuppressWarnings("unchecked")
+  private void fetchAndWriteLineage(Set<String> entityIds, FileWriter fw,
+                                    ClientConfig config) throws IOException {
+    HttpURLConnection connection = createHttpConnection(config, entityIds);
+
+    InputStream stream = connection.getInputStream();
+
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    mapper.configure(JsonParser.Feature.ALLOW_COMMENTS, true);
+    ArrayList entities = (ArrayList) mapper.readValue(stream, Object.class);
+
+    Iterator it = entities.iterator();
+    while (it.hasNext()) {
+
+      Map<String, Object> m = (Map<String, Object>) it.next();
+
+      String id = (String) m.get("id");
+      Map<String, Object> children = (Map<String, Object>) m.get("children");
+
+      if (children != null) {
+        for (Map.Entry<String, Object> entry : children.entrySet()) {
+          ArrayList<String> childrenIds = (ArrayList<String>) entry.getValue();
+
+          for (String childId : childrenIds) {
+            fw.write(String.format("%s, %s, %s%n", id, childId, entry.getKey()));
+          }
+        }
+      }
+    }
+
+    IOUtils.close(connection);
+  }
+
+  private String constructEntityIdsAsCSV(Set<String> entityIds) throws UnsupportedEncodingException {
+
+    if (entityIds.size() == 0) {
+      return "";
+    }
+
+    Set<String> ids = Sets.newHashSet();
+    for (String id : entityIds) {
+      ids.add("\"" + id + "\"");
+    }
+
+    Joiner join = Joiner.on(",");
+    return join.join(ids);
+  }
+
+  private HttpURLConnection createHttpConnection(ClientConfig config, Set<String> entityIds)
+      throws IOException {
+    String navUrl = config.getNavigatorUrl();
+    String serverUrl = navUrl + (navUrl.endsWith("/") ? LINEAGE_EXPORT_API : "/" + LINEAGE_EXPORT_API);
+    String queryParamString = String.format(LINEAGE_EXPORT_API_QUERY_PARAMS,
```

Comment:
Sure I could do that but then within the custom responseextractor, I would be again using the mapper to deserialize the inputStream into a collection of entities (correct me if that's not so). I think the use of RestTemplate would make sense if I create a class backing the response from the API.